drgn_tools/task.py

```diff
@@ -161,3 +170,91 @@ def task_cpu(task: drgn.Object) -> int:
     if has_member(task, "cpu"):
         return task.cpu.value_()
     return task_thread_info(task).cpu.value_()
+
+
+def format_nanosecond_duration(nanosecs: int) -> str:
+    """
+    :returns: conversion of nanoseconds to [dd hh:mm:ss.ms] format
+    """
+    secs = nanosecs_to_secs(nanosecs)
+    dd, rem = divmod(secs, 86400)
+    hh, rem = divmod(rem, 3600)
+    mm, secs = divmod(rem, 60)
+    return "%02ld %02ld:%02ld:%.3f" % (dd, hh, mm, secs)
```

Comment:
For the seconds component, you should use `06.3f` as the format argument.
- 0 is the padding character
- 3 is the precision, the number of digits after the decimal place to show
- 6 is the total width of the field, including the `.` - using 6 ensures that you always get `xx.xxx`
```suggestion
    return "%02ld %02ld:%02ld:%06.3f" % (dd, hh, mm, secs)
```